Source/WebCore/PAL/pal/system/ios/UserInterfaceIdiom.mm

```diff
@@ -44,11 +48,19 @@ bool currentUserInterfaceIdiomIsSmallScreen()
     return s_currentUserInterfaceIdiom.load() == UserInterfaceIdiom::SmallScreen;
 }
 
-bool currentUserInterfaceIdiomIsVision()
+bool currentUserInterfaceIdiomIsVisionOrVisionLegacy()
+{
+    if (!s_currentUserInterfaceIdiom.load())
+        updateCurrentUserInterfaceIdiom();
+    auto idiom = *s_currentUserInterfaceIdiom.load();
+    return idiom == UserInterfaceIdiom::Vision || idiom == UserInterfaceIdiom::VisionLegacy;
```

Comment:
Optional nit: you don't need to unwrap the value because of this `operator==` overload.

```cpp
template< class T, class U >
constexpr bool operator==( const optional<T>& opt, const U& value );
```

Reply:
but then wouldn't we be performing two atomic loads?

Reply:
no, I think it's still just one load (we'd load into a local optional<uint...>). But I don't love the thought of it :P